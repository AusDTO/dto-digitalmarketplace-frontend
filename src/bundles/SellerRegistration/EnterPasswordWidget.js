import React from 'react'
import { Provider } from 'react-redux'
import RegisterComponent from '../../RegisterComponent'

import createStore from './redux/create'
import EnterPasswordForm from './components/EnterPasswordForm'

export const EnterPasswordWidget = (props) => {
  const store = createStore(props)
  return (
    <Provider store={store}>
      <EnterPasswordForm />
    </Provider>
  )
}

export default new RegisterComponent({ enterpassword: EnterPasswordWidget })
