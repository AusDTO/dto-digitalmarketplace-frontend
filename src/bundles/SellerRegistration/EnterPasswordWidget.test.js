// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react';
import {EnterPasswordWidget} from './EnterPasswordWidget';
import renderer from 'react-test-renderer';

test('EnterPasswordWidget renders correctly', () => {
  const component = renderer.create(
    <EnterPasswordWidget />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
