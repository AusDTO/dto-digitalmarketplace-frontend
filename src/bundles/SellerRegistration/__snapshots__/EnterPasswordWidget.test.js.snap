// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterPasswordWidget renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-push-2 col-sm-8 col-xs-12"
  >
    <div
      className="layout"
    >
      <header>
        <h1>
          Create a password
        </h1>
        <p>
          To finish creating your account you need to create a password.
        </p>
      </header>
      <article
        role="main"
      >
        <form
          action={undefined}
          id="enterpassword"
          method="post"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="field"
          >
            <div
              className="question-heading"
            >
              Email
            </div>
            <p />
          </div>
          <div
            className="field"
          >
            <label
              className="question-heading"
              htmlFor="password"
            >
              Password
            </label>
            <p
              className="hint"
              id="password-hint"
            >
              At least 10 characters.
            </p>
            <input
              aria-describedby="password-hint"
              className=""
              disabled={false}
              id="password"
              maxLength={undefined}
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              pattern={undefined}
              type="password"
              value=""
            />
          </div>
          <input
            role="button"
            type="submit"
            value="Complete account creation"
          />
        </form>
      </article>
    </div>
  </div>
</div>
`;
